### PR TITLE
Add a flutter-attach entry point for fuchsia

### PR DESCRIPTION
**Pull request by jonahwilliams**
Port the script to run flutter attach from the fuchsia tree back into flutter.

~~Still needs to be manually verified in-tree.~~
Verified that hot reload works with these changes patched into the local flutter checkout

### CODE REVIEW

**Comment by jonahwilliams** [comment]
cc @abarth  @tvolkert

File: packages/flutter_tools/fuchsia_attach.dart [hunk 1]
```diff
@@ -0,0 +1,128 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter_tools/fuchsia_attach.dart]: ...woops

File: packages/flutter_tools/fuchsia_attach.dart [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  ..addOption('build-dir', help: 'The fuchsia build directory')
+  ..addOption('dart-sdk', help: 'The prebuilt dart SDK')
+  ..addOption('target', help: 'The GN target to attach to')
+  ..addFlag('verbose', negatable: true);
+
+// Track the original working directory so that the tool can find the
+// flutter repo in third_party.
+String originalWorkingDirectory;
+
+Future<void> main(List<String> args) async {
```
Comment (tvolkert) [inline packages/flutter_tools/fuchsia_attach.dart]: This can be just `void` now

Comment (jonahwilliams) [reply packages/flutter_tools/fuchsia_attach.dart]: We have the avoid_void_async lint enabled.

File: packages/flutter_tools/fuchsia_attach.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter_tools/fuchsia_attach.dart]: Fixed

File: packages/flutter_tools/fuchsia_attach.dart [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  final ArgResults argResults = parser.parse(args);
+  final bool verbose = argResults['verbose'];
+  final String target = argResults['target'];
+  final List<String> targetParts = _extractPathAndName(target);
+  final String path = targetParts[0];
+  final String name = targetParts[1];
+  final File dartSdk = fs.file(argResults['dart-sdk']);
+  final String buildDirectory = argResults['build-dir'];
+  final File frontendServer = fs.file('$buildDirectory/host_x64/gen/third_party/flutter/frontend_server/frontend_server_tool.snapshot');
```
Comment (tvolkert) [inline packages/flutter_tools/fuchsia_attach.dart]: This should probably be controlled by a flag

Comment (tvolkert) [reply packages/flutter_tools/fuchsia_attach.dart]: scratch that... then all of these would be as well.  I guess this is fine to be tightly coupled to Fuchsia tree structure.

File: packages/flutter_tools/fuchsia_attach.dart [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  final File flutterPatchedSdk = fs.file('$buildDirectory/flutter_runner_patched_sdk');
+  final String packages = '$buildDirectory/dartlang/gen/$path/${name}_dart_library.packages';
+  final String outputDill = '$buildDirectory/${name}_tmp.dill';
+
+  // TODO(jonahwilliams): running from fuchsia root hangs hot reload for some reason.
+  // switch to the project root directory and run from there.
+  originalWorkingDirectory = fs.currentDirectory.path;
+  fs.currentDirectory = path;
+
+  // Check for a package with no lib directory.
```
Comment (tvolkert) [inline packages/flutter_tools/fuchsia_attach.dart]: with _a_ lib directory

File: packages/flutter_tools/fuchsia_attach.dart [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+  ];
+  if (verbose) {
+    command.add('--verbose');
+  }
+  Cache.disableLocking(); // ignore: invalid_use_of_visible_for_testing_member
+  await runner.run(
+    command,
+    <FlutterCommand>[
+      _FuchsiaAttachCommand(),
+      _FuchsiaDoctorCommand(),
```
Comment (tvolkert) [inline packages/flutter_tools/fuchsia_attach.dart]: Add a comment explaining why this is needed

**Review by tvolkert — APPROVED** [review]

**Pull request merged**